ipamd/ipamd.go

```diff
@@ -700,21 +706,22 @@ func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
 // 2) set up linux ENI related networking stack.
 // 3) add all ENI's secondary IP addresses to datastore
 func (c *IPAMContext) setupENI(eni string, eniMetadata awsutils.ENIMetadata) error {
-	// Have discovered the attached ENI from metadata service
-	// add eni's IP to IP pool
-	err := c.dataStore.AddENI(eni, int(eniMetadata.DeviceNumber), eni == c.awsClient.GetPrimaryENI())
-	if err != nil && err.Error() != datastore.DuplicatedENIError {
-		return errors.Wrapf(err, "failed to add ENI %s to data store", eni)
-	}
-
 	ec2Addrs, eniPrimaryIP, err := c.getENIaddresses(eni)
 	if err != nil {
 		return errors.Wrapf(err, "failed to retrieve ENI %s IP addresses", eni)
 	}
 
+	// Add the ENI to the datastore
```

Comment:
@mogren I think I am missing something. Why did you switch the order?

Reply:
@tiffanyfay Thanks, good question! I tested this a few times, and if `c.getENIaddresses()` fails, it means that `c.awsClient.DescribeENI()` that does an actual call to EC2 failed. That in turn means we don't get the list of IP for that ENI, so there is no point in adding the ENI to the datastore just to have it be [removed in the next reconcile loop](https://github.com/aws/amazon-vpc-cni-k8s/blob/c6f082ed6fed7cc9b59e5365d97dcc36b9f0716b/ipamd/ipamd.go#L947).

Reply:
Ahh ok, I get it--I was wondering if ec2Addrs would be outdated, but it's just putting the info in the datastore.